feature-libs/asm/root/services/csagent-auth.service.ts

```diff
@@ -52,14 +53,14 @@ export class CsAgentAuthService {
         password
       );
       // Start emulation for currently logged in user
-      let customerId: string;
+      let customerId: string | undefined;
       this.userService
         .get()
         .subscribe((user) => (customerId = user?.customerId))
         .unsubscribe();
       this.store.dispatch(new AuthActions.Logout());
 
-      if (Boolean(customerId)) {
+      if (customerId !== undefined && userToken !== undefined) {
```

Comment:
Different fix

Reply:
The shorter version seems fine as the edge cases addressed by typeof are mitigated.